### PR TITLE
Rename classes and functions related to eigenstrain calculation

### PR DESCRIPTION
**Pull request by bwspenc**
ref #7815

### CODE REVIEW

**Comment by bwspenc** [comment]
@dschwen Can you take a look at this?  I think it's ready to go. I will need to submit some application patches to address the API incompatibilities.

File: modules/tensor_mechanics/include/materials/ComputeEigenstrainBase.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef COMPUTEEIGENSTRAINBASE_H
+#define COMPUTEEIGENSTRAINBASE_H
+
+#include "Material.h"
+#include "RankTwoTensor.h"
```
Comment (dschwen) [inline modules/tensor_mechanics/include/materials/ComputeEigenstrainBase.h]: Please use a forward declaration here and move the include into .C file

File: modules/tensor_mechanics/include/materials/ComputeThermalExpansionEigenstrain.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef COMPUTETHERMALEXPANSIONEIGENSTRAIN_H
+#define COMPUTETHERMALEXPANSIONEIGENSTRAIN_H
+
+#include "ComputeEigenstrainBase.h"
+
+class ComputeThermalExpansionEigenstrain;
+class RankTwoTensor;
```
Comment (dschwen) [inline modules/tensor_mechanics/include/materials/ComputeThermalExpansionEigenstrain.h]: same here

File: modules/tensor_mechanics/include/materials/ComputeThermalExpansionEigenstrain.h [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+{
+public:
+  ComputeThermalExpansionEigenstrain(const InputParameters & parameters);
+
+protected:
+  virtual void computeQpEigenstrain();
+
+  const VariableValue & _temperature;
+  const Real & _thermal_expansion_coeff;
+  MaterialProperty <RankTwoTensor> & _thermal_expansion_tensor;
```
Comment (dschwen) [inline modules/tensor_mechanics/include/materials/ComputeThermalExpansionEigenstrain.h]: no space before `<`

**Review by dschwen — CHANGES_REQUESTED** [review]
Please also rename the .md files in the new documentation system and adjust the pages.yml files.

File: modules/tensor_mechanics/include/materials/ComputeThermalExpansionEigenstrain.h (hunk 2 above)
Comment (bwspenc) [inline modules/tensor_mechanics/include/materials/ComputeThermalExpansionEigenstrain.h]: Huh? It's already a forward declaration here.

Comment (dschwen) [inline modules/tensor_mechanics/include/materials/ComputeThermalExpansionEigenstrain.h]: 🍻 😵 🍻

**Comment by bwspenc** [comment]
Ok, I addressed all of the issues.

**Review by dschwen — APPROVED** [review]

**Comment by bwspenc** [comment]
Latest version is a minor tweak to fix the build issues -- there were a couple of derived classes that had member `RankTwoTensor` objects that got messed up by the change to forward declaration of that in the base class.

**Comment by dschwen** [comment]
```
tensor_mechanics:generalized_plane_strain.generalized_plane_strain_squares: Deprecated Object: ComputeThermalExpansionEigenStrain
```

**Review by dschwen — CHANGES_REQUESTED** [review]
Remove use of deprecated object in test.

**Comment by bwspenc** [comment]
Fixed that test with the deprecated object.

**Review by dschwen — APPROVED** [review]

**Comment by dschwen** [comment]
I'll go ahead and merge this regardless of the ongoing (unrelated) Yak breakage. @bwspenc is working on the Bison and Marmot PRs now.

**Comment by permcody** [comment]
Sigh... This is a bad idea in general. Integration testing is important and if we have multiple PRs that require patches to applications at the same time we start to lose control on testing those changes individually. We now five PRs merged two of which cause "expected" breakage. I'd prefer that we space this out a little more in the future. We now have no way of getting submodule updates of these effects separately.

**Pull request merged**